docs/rules.md

```diff
@@ -46,6 +46,34 @@ hook_helloworld:
 
 This rule will inject `MyHookBefore` at the start of the `Example` function in the `main` package, and `MyHookAfter` at the end. The hook functions are located in the specified `path`.
 
+**Generic Function Support:**
```

Comment:
The documentation needs to be updated to state that SetParam and SetReturnVal cannot be used with generic hook functions.